clang/test/Driver/split-debug.c

```diff
@@ -11,7 +11,6 @@
 // NOINLINE-NOT: "-fsplit-dwarf-inlining"
 // SPLIT-NOT:  "-dumpdir"
 // SPLIT:      "-debug-info-kind=constructor"
-// SPLIT-SAME: "-ggnu-pubnames"
```

Comment:
has this lost coverage for non `-glldb` configurations?

(like would we still have test failures if we had a bug where `-ggnu-pubnames` wasn't passed on Linux when using `-gsplit-dwarf`?)

Reply:
Sorry not sure I follow.
As discussed in original diff the idea is if -gsplit-dwarf with -glldb is used the -ggnu-pubnames wouldn't be passed in. Since LLDB doesn't benefit from it, and it just uses space.
A fix from original is if someone does specify -ggnu-pubnames it will be generated. I added a test for it on linux side. There is also a test on mac side that used -gsplit-dwarf with -ggnu-pubnames. On mac LLDB tunning is the default. So equivalent of passing -glldb on linux.

Reply:
> As discussed in original PR, #82840 , the idea is if -gsplit-dwarf with -glldb is used the -ggnu-pubnames wouldn't be passed in. Since LLDB doesn't benefit from it, and it just uses space.

Yep, I'm with you on all of that.

> A fix from original is if someone does specify -ggnu-pubnames it will be generated.

Sure.

> I added a test for it on linux side.

Added a test for what on the linux side? And where is that test?

> There is also a test on mac side that used -gsplit-dwarf with -ggnu-pubnames.

Right, I see that test here in this `split-debug.c` file at the end. Checking for the absence of `-ggnu-pubnames` if `-gsplit-dwarf -g -glldb` is passed, and the presence if `-gsplit-dwarf -g -glldb -ggnu-pubnames` is passed. So I'm with you there.

But what i don't understand is that, with the removal of the `// SPLIT-SAME: "-ggnu-pubnames"` line above - what tests that, for instance `-gsplit-dwarf -g -ggdb` does use `-ggnu-pubnames`? (said another way: If we removed the any code that added `-ggnu-pubnames` when `-gsplit-dwarf` is specified, which lit tests would fail?)

Reply:
Oh I see!
Thanks for clarification. 
That check should be there. I think it snuck in when I tried to re-apply a patch internally and somehow got merged with original version of the first PR.

Reply:
Nah, I think that's probably fine without an explicit `-ggdb` test - though with the test as-is, wouldn't it fail on macos machines, where they'll implicitly default to `-glldb` and so `-ggnu-pubnames` won't get passed, so the check on line 14 of split-debug.c will fail? So maybe that has to be removed from there and only checked via explicit tuning flags, in which case adding a `-ggdb` test would be necessary (though it could share the CHECK lines from the `-glldb -ggnu-pubnames` test)

Reply:
At least on my mac it passed. With that check.
I double checked, and this is what I see:
./bin/clang -### -c -target x86_64 -gsplit-dwarf -g /Users/ayermolo/local/llvm-project/clang/test/Driver/split-debug.c
"-dwarf-version=5" "-debugger-tuning=gdb" "-ggnu-pubnames"
./bin/clang -### -c -gsplit-dwarf -g /Users/ayermolo/local/llvm-project/clang/test/Driver/split-debug.c
"-dwarf-version=4" "-debugger-tuning=lldb"